### PR TITLE
Massive Combined Medbay / Science / Escape / Southern Maint Remap

### PR DESCRIPTION
**Pull request by FreeStylaLT**
### DISCLAIMER
I am fully aware this PR is not going to be merged for a long while. The intention here is to get it further tested, tweaked and perfected for when the map queue opens up.

## What Does This PR Do
After some grueling years-long analysis a conclusion was made -- the current Medbay is too cramped, small, did not have any antag opportunities and had Genetics in it. This PR's primary focus is to fix that. However, this was not possible without expanding Medbay, and consequently the rest of the station, making a Science, Escape and Scimaint remaps unavoidable.

## Why It's Good For The Game
Assuming my mapping worked out, this change should make Medbay far more higher-pop compatible, increase potential antag activity, both with layout changes in Medbay itself and to the surrounding maintenance areas.

## BIG IMAGE (DIFFBOT):
image

## MEDBAY (IN-GAME)
The overall treatment areas have been *greatly* expanded, with the addition of three dedicated treatment rooms.

### LOBBY / RECEPTION / CHEMISTRY
image

The lobby has been shrunk down overall, as it took up unnecessary space. Chemistry swapped places for convenience of use. Reception expanded, break room integrated into it. This was to consolidate superfluous space into a single room, while keeping it looking as natural as possible, to allow for more treatment space.

### CMO / CLONING
image
CMO now directly overlooks cloning and secure storage. Genetics is gone from Medbay. Cloning has a dedicated corpse chute directly linked to the Morgue. Robotics has this too.

Another notable addition is the mini-maint area, containing all of the oxygen supply for cloning cryo, along with most APCs adjacent to the area. **Engineers AND Doctors have access to that area from Medbay proper (Not to / from cloning or storage. That is doctor only.) This applies to the other supply maint by cryo.**

### SURGERY
image
An actual, large lobby and, as requested, two distinct surgery theatres -- one transparent, another opaque. There is a third surgery room in maint that is abandoned, but can be restored.

**The AI has no vision of the private surgery room and the surgery storage room. This is intentional.**

### STORAGE
image
Greatly expanded and consolidated. The distinction between secure storage and supplies has been made clear. Doctors still have access to both. Hardsuits are now locked behind SSUs.

### VIROLOGY
image
Shrunk down further, as it is only meant to support one person and some monkeys, now no longer requires an entire secondary hallway, merely a mini-lobby.

### MISC
Besides the corpse chute, another addition was the ability to make the small treatment rooms private, via electrochromic windows:
ezgif-3-5b46da065a3a

## SCIENCE (IN-GAME)
While unwanted and arguably never needed, Science was also remapped. The main focus was on moving Genetics and streamlining the superfluous hallways, otherwise keeping it as intact and the same as possible.

### LOBBY / R&D
image
A combined R&D / Robotics lobby was added. R&D has been fancied up and made slightly bigger.

### ROBOTICS
image
Increased in size (as per feedback), the mech bay and actual lab have been distinguished. Mech bay is now, at round-start, public access, unless Robo-dudes decide to close the external shutters. A dedicated surgery room was also added, though the RD, if present, can take a peek at who you're cutting exactly, buckaroo. **The AI has no vision of the surgery room. This is intentional.** :eyes:

### RD OFFICE / SERVER ROOM
image
Since the RD is the sole person with access to the server room, it only made sense to have him be adjacent to it. Otherwise, merely an aesthetic change.

### GENETICS / EXPERIMENTOR
image
Moved to Science and shrunk overall. No other real notable changes. Genetics has been plagued with issues coming from its combined oversight. Their main job involves pressing buttons and they have rarely been seen interacting with the cloning equipment.

### XENOBIO
image

Made horizontal and streamlined to be more narrow. Allows for easier adding (or removing) chambers.

### RAGE CAGE
image
A simple half-bbq half-rage cage, to compensate for the late removal of the tiny boxing ring in Dorms.

### ABANDONED MEDICAL WING
image
In Medbay maint, there is now this dark, abandoned wing. It has a non-functioning chemistry that *can* be rebuilt, same for cryo and some loot, depending on RNG.

### MAINT AIR SUPPLY ROOM
image
A big one. This room now supplies air to the entire southern maint block through the cyan pipes.

## Changelog
:cl: FreeStylaLT
tweak: Mapped a lil' bit of medbay and science, some maints too i guess idk.
/:cl:

### CODE REVIEW

**Comment by warriorstar-orion** [comment]
This is hugely exciting, for real.

I don't suppose xenobio can be pulled out of the station and into a satellite-like area, similarly to Meta? https://tgstation13.org/wiki//images/8/80/MetaStation.png

That isolation does sort of emphasize the dangers of xeno, and opens up that space in maints.

**Comment by AffectedArc07** [comment]
image
:angryeyes:

**Comment by Mitchs98** [comment]
Overall like 90% of this. I have no real critiques other than I would _really_ rather not have it be possible to just emag/break into surgery through maint so easily VIA the door. It was my main gripe about the medbay before this one and I'd rather it not return.

Bit awkward having Chemistry on the left again but not a major concern.

Overall, quite like the remap.

**Comment by Carthusia** [comment]
I love the revamp, but one thing that jumps out at me

https://i.gyazo.com/2868970b33243cc3bb2c34996db977e3.png
Public Pod Parking: Perhaps expand this to be a bit bigger? Its only really got enough space for 1, maybe 2 pods if your really cramming and a mechanic can easily pump out 10-20 of them on a good shift, not saying make it big enough to hold that many, but a decent amount of parking space would be nice. Perhaps also remove the reinforced walls around it to allow for player expansions to its area as well.

Never really been sure why that pod hanger has reinforced walls, when every area directly around it does not. Perhaps condense that O2 locker room into the pod parking zone and expand it length wise across the top.

File: code/game/jobs/job/science.dm [hunk 1]
```diff
@@ -121,3 +121,36 @@
...
+
+/datum/job/geneticist
+	title = "Geneticist"
+	flag = JOB_GENETICIST
+	department_flag = JOBCAT_MEDSCI
+	total_positions = 2
+	spawn_positions = 2
+	is_medical = 1
+	supervisors = "the chief medical officer and the research director"
+	department_head = list("Chief Medical Officer", "Research Director")
```
Comment (Kyep) [inline code/game/jobs/job/science.dm:133]:
```suggestion
	supervisors = "the research director"
	department_head = list("Research Director")
```

File: code/game/area/ss13_areas.dm [hunk 2]
```diff
@@ -676,6 +676,13 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 	power_light = 0
 	power_environ = 0
 
+/area/maintenance/abandonedrobotics
+	name = "Maintenance Robotics"
+	icon_state = "abandonedrobotics"
+	power_equip = 0
+	power_light = 0
+	power_environ = 0
```
Comment (AffectedArc07) [inline code/game/area/ss13_areas.dm:684]:
```suggestion
	power_equip = FALSE
	power_light = FALSE
	power_environ = FALSE
```

File: code/game/jobs/job/science.dm (hunk 1 above)
Comment (Kyep) [inline code/game/jobs/job/science.dm:133]:
```suggestion
	is_science = TRUE
	supervisors = "the research director"
	department_head = list("Research Director")
```

**Comment by S34NW** [comment]
image
Perhaps the wiring here could be made a little simpler? It's uncommon to see curved branches like this.

I'm also like that medbay and science are not powered by multiple routes now, makes life slightly easier for antags wishing to mess with the power in those areas.

**Comment by Bkulli-VI** [comment]
ViVi here, did a shift as the RD, overall like the changes. but a million small tweeks.  RnD only started with 4 sheets of glass which is a bit of a change at the start.  Also Robotics and RnD lack a windoor at the desk. (only about 4 people broke in this time, slow shift).  Also had to teach myself how to set up the server so that other computers would sync to the RnD desk.  Also someone reported that toxins didn't survive a few test but I didn't confirm that well myself. 

Other stuff I probably forgot to write down, but overall a fan. Robotics needed the room for sure

**Comment by Spacemanspark** [comment]
My only issue is with robotics and I've already brought it up with free, but I'll stick it here as well; there really shouldn't be any windows on the north side. People can go to the desk on the right in the new lobby if they need something, otherwise you'll just have a massive crowd watching over you and screaming for implants.
Sure, there are shutters, but anytime I touch them it usually ends up with people whining over common and/ or the RD walking in, lifting the shutters without talking to anyone and saying "don't do that". It's fairly pointless to me.

**Comment by Pumpkin312** [comment]
As an assistant virtuoso, i really love this. maints has all the things that encourage bored people to build amazing things.
only thing i didn't like was that even though xenobio looks cute its very cramped so its gonna suck when people start using certain extracts or when you have two xenobiologists.

**Comment by KozackWolf** [comment]
My only real issue with the layout is genetics needs to be closer to cloning. Cloning needs quick access to Clean S.E. and humanized animals. Currently. Rather than going through one door from cloning you have to go through 5 doors, Currently unsure if medical even has access to genetics doors as I didn't play the round.

Other than that, I like the setup. It'll take getting used to.

If you fix genetics location, I'd love this to be merged.

**Comment by Mitchs98** [comment]
> My only real issue with the layout is genetics needs to be closer to cloning. Cloning needs quick access to Clean S.E. and humanized animals. Currently. Rather than going through one door from cloning you have to go through 5 doors, Currently unsure if medical even has access to genetics doors as I didn't play the round.
> 
> Other than that, I like the setup. It'll take getting used to.
> 
> If you move genetics location, I'd love this to be merged. Try moving it above virology.

I didn't even think about this, but this is also a major concern for chemistry too.

**Comment by lolzyking** [comment]
I have a major issue with this PR https://i.gyazo.com/3a7b9c864dc379fc88b7e1cb261c58a8.png  as you can see the welding tank is no longer directly south of the door. https://i.gyazo.com/eefacb03047e7cb4a7fa1c2d18a20613.png
this is a very important death trap.

**Comment by moxian** [comment]
The garbage chute at cloning south side is awful. It makes it very awkward for two people to operate the cloning machinery at once (not "needed" ideally, but happens extremely often in practice, especially when mentoring newer people), and anyone going through the south door *has* to shuffle up to 3 people at the machine

Please strongly consider doing something like 
clo2
or even 
clo3


@KozackWolf 

> Cloning needs quick access to Clean S.E. 

Never have I had a use for clean S.E as a doctor.

> and humanized animals

In the remap, the CMO office has monkey cubes. The fact that currently those are stored (only) in genetics is happenstance, it doesn't need to be that way. Medical doesn't need to get cubes from genetics - they can just as well get those from, say, chef. Or we can store them somewhere in medbay, and request geneticists to grab those from there.
Honestly, I'd prefer if those were in medical storage (in addition to what genetics has), but them being in CMO office is fine as well.

**Comment by KozackWolf** [comment]
Well, not everyone is human, so you'd need EVERY type of cube in medical, also, it's not about the cubes, it's about genetics easily humanizing them. A medical person getting to genetics to get a humanized animal easily is important.

Humanizing, you usually use a Clean S.E.

**Comment by TDSSS** [comment]
Using humanized monkeys feels like a more advanced procedure and not a basic thing the medbay should be optimized for, personally.

**Comment by tiggypuff** [comment]
I've got two issues with the surgery rooms.
First, the private surgery room strikes me as odd. Every treatment room has some form of optional privacy (electochromic windows or shutters), however, this one room does not. While I understand the necessity of being able to do surgery from a gameplay side, I can't imagine an ingame reason to break the pattern of optional privacy with this one room. Also, I can imagine that a lot of doctors would barge in on a ongoing surgery since they can't see from the outside if the room is actually in use. This might even result in the room being less desirable for doing surgery in secret. I'd rather see this room being outfitted with shutters and/or electrochromic glass too.
My other issue is with the non-private surgery room. Only a small issue, but I'd swap the positions of the surgery table and the computer. That way you won't have to drag the patient around the computer to access the table, and the surgeon can stand in a cardinal direction next to the surgery table while being able to access all three tables.

All in all, I like the Medbay remap. The additional treatment room and the corpse chute in Cloning should help to reduce the clogging in Medbay during busy shifts.

**Comment by Dashsora** [comment]
Overall, after really exploring it all - it feels pretty damn great. I especially love the disposals tricks for toxins and the morgue - although I did hear some concern about putting trash in the morgue's bin.. If there isn't a windoor for medical staff guarding it there probably should be (I don't remember exactly anymore).

My only complaint is that I really feel that maint, at some places, has a little bit of an overabundance of doors. I totally get that the intent was to make it feel maze but remember that maint doors do restrain ID-less and cuffed movement, and borgs/the AI has absolute dominion on them, unless you spend your time disabling AI wires, not to mention the risk of power failures and suddenly all the people without a crowbar will be stuck in XYZ room. They have a pretty important mechanical signification.

One of the main reason I prefer Paradise's station over most other is because it's very clean. I feel like a lot of other stations have stuff all over the place, and very cluttered/restricted hallways. With so many doors, those places really feel like that to me. If you look at all other maint areas on station, they all have very clearly defined hallways with airlocks separating different areas or rooms on the side, and I miss a little bit of that in some places.

Here are the three biggest spots I have in mind : 

 - The intersection right outside of robotics/upper science and virology/morgue, right above the little maint bar. Right now, it's separated into two rooms, the hallway leading to main hallway, and a ring-around of doors around a central wall.
In this example : [https://gyazo.com/337aa2d70c811aa3fdecf1bea9851810](url)
I merged the two rooms, but I honestly wouldn't mind it being one with the whole hallways leading up. I liked what you had there on the OP's full map more.

 - The hallways south of the first example, north and leading to the entrance of xenobiology. Right now it's a full on straight hallways cornering around two other maint rooms, but just sectionned off by three doors for no discernabe reason. It really feers like airlocks for the sake of airlocks, and is very frustrating to just walk through.
In this example : [https://gyazo.com/5db9a67217903c75c4e1b6995f7b5482](url)
I'd suggest remove at least the two red doors, maybe even the blue one - they don't add much and will be yet another buncha rooms to get stuck in until someone walks by. Could replace one of the reds with a barricade, and leave the little wall indentations, or just girders.

 - The three rooms right south of scichem's testing room. Right now it's a hallways separated by another ring around of doors separating it into three fairly meaningless rooms. (As an aside, I thoroughly like the cheeky WF tank right behind the scichem test room wall)
In this example : [https://gyazo.com/ccee39c035004a5b9d940bcde747b8d2](url)
I'd suggest removing what's tagged red, and replacing the blue one with a barricade. Connecting the eastern side as a hallways to the abandonned robotics lab, and, when the barricade is cleared, all the way to the xenobiology entrance. I wouldn't even be against removing the southern-most door. 

I feel like cleaning up those three areas slightly wouldn't detract at all from the maze-like feelings that's desired, but would really help with freedom of movement and reducing clutter, making it a little more in line with what's the norm with the rest of the station.

**Comment by TDSSS** [comment]
Maybe the doors leading from one maint area to another should be access-less, with only the ones leaving out of maint needing access.

**Comment by Adrer** [comment]
Just two minor remarks that I noticed while playing the remap as AI
1. There's holopads missing in a few locations that I'd really like to have holopads, namely the morgue, chemistry, and the maybe even the medbay reception.
2. A camera + camera console setup in scichem (test chamber) like pre-remap would be lovely.

**Comment by KozackWolf** [comment]
After more test merges, still don't enjoy the fact that genetics is so far away from medical.

**Comment by S34NW** [comment]
I think genetics still needs an air alarm

**Review by Kyep — APPROVED** [review]

**Comment by Spacemanspark** [comment]
Please, please remove those front windows from robotics. Those being there is just asking for trouble lmao

Other than that, really good stuff.

**Comment by Carthusia** [comment]
Id agree, the windows at robotics arnt really that required, unless they can be fully shuttered without locking down the entire bay

**Comment by FreeStylaLT** [comment]
PR has been brought up to speed and edited with the **astounding** amount of insightful feedback I was provided by @Vulkoras

**Comment by SabreML** [comment]
The big central hallway in medbay feels a bit barren as it is now, I think it could do with some more decoration around the centre at least.
**Current/Idea**
current idea

It works just fine as it is of course, but it can't hurt to make it a bit more welcoming.

**Comment by S34NW** [comment]
image
Any reason these walls are reinforced in chapel? pod may below is not

**Comment by NotDhu** [comment]
While genetics has always felt more like a part of science than medical to me, medical not having access to clean SEs or humanized animals can sometimes slow down treatment a lot. 

The 'private operating room' with no windows also feels really awkward. The shutters/window tint buttons already exist for privacy, the walls honestly just seem really out of place. The lack of any windows just leads to other doctors walking in on the surgery anyways, which just crowds the OR and can sometimes lead to infections.

If the intention was to give antagonists an easy way to privately murder people via surgery, the added abandoned operating room in maintenance already does that and is located in roughly the same area. I don't really understand the need for this.

File: code/modules/mob/living/simple_animal/hostile/retaliate/kangaroo.dm [hunk 3]
```diff
@@ -21,6 +21,7 @@
 	atmos_requirements = list("min_oxy" = 5, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 2, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
 	move_to_delay = 4 // at 20ticks/sec, this is 5 tile/sec movespeed, about the same as a 'fast human'.
 	speed = -1 // '-1' converts to 1.5 total move delay, or 6.6 tiles/sec movespeed
+	var/melee_damage_special = 60
```
Comment (AffectedArc07) [inline code/modules/mob/living/simple_animal/hostile/retaliate/kangaroo.dm:24]: Undocumented change?

Comment (FreeStylaLT) [reply code/modules/mob/living/simple_animal/hostile/retaliate/kangaroo.dm:24]: No, yes, sort of. I added the variable so I could tweak it for subtypes / instances like Wally and be able to nerf the Kangaroo's (rather insane) base damage value for its special attack. 

The number was originally hardcoded and unmodifiable. I didn't change the value itself (Normal Kangaroos still wallop for 60 damage), only added the variable.

File: code/modules/mob/living/simple_animal/hostile/retaliate/kangaroo.dm [hunk 4]
```diff
@@ -48,8 +49,8 @@
 	// ... but, every attack_cycles_max attacks on a living mob, do a powerful disemboweling kick instead
 	attack_cycles = 0
 	attacktext = "VICIOUSLY KICKS"
-	melee_damage_lower = 60
```
Comment (FreeStylaLT) [inline code/modules/mob/living/simple_animal/hostile/retaliate/kangaroo.dm:51]: Case in point. The values were hardcoded for the special attack.

File: code/modules/research/experimentor.dm [hunk 5]
```diff
@@ -543,7 +544,7 @@
 		usr << browse(null, "window=experimentor")
 		return
 	else if(scantype == "search")
-		var/obj/machinery/computer/rdconsole/D = locate(/obj/machinery/computer/rdconsole) in oview(3,src)
+		var/obj/machinery/computer/rdconsole/D = locate(/obj/machinery/computer/rdconsole) in oview(range_search,src)
```
Comment (FreeStylaLT) [inline code/modules/research/experimentor.dm:547]: Same reasoning as with Kangaroos here. Added a variable to allow for easier in-map tweaking of consoles.

**Comment by cloutcode** [comment]
major medbay overhaul... still no medical records console in the morgue... we live in a SOCIETY that maligns coroners...

**Comment by AffectedArc07** [comment]
image
image

**Comment by S34NW** [comment]
xenobio to smol ples expand mor

**Comment by FreeStylaLT** [comment]
:trollface:

**Comment by ZanderDoggo** [comment]
walls in chem should be reinforced like they are on normal map (I think they are?)

**Comment by BakKeener** [comment]
After being reminded about this PR and being a long time Medical Main, I decided to take another look at the remap to see what I can bring to the table. I like Genetics being moved to Science entirely due to them having almost nothing to do with Medical as the player base in genetics hardly does anything with cloning and indeed usually only pushes buttons sitting in their room.

I believe OR-1 which is the closed surgery room should have windows but have that tinted window feature you displayed earlier to prevent doctors from stumbling in when OR-2 is occupied while someone is doing surgery. Another nitpick that I found is that I don't find any reason to have a window to virology in maint as shown here. 
image
Virology should be the most sterile environment for the nature of work that occurs inside and having a window that leads into maintenance is a safety hazard that I believe NT wouldn't have placed there. 

I do however like how Virology is isolated again which allows for better antag plays.
Morgue having more space and slabs is very pleasant, needs at least 1 more light bulb as it's too dark even with night shift lighting off. I also like the expansion of space for chemistry as it looks like it's coming back to it's roots from before.  I don't know how I feel about most the APCs being put in the medical maint rooms as it makes it harder for engineering to fix? I don't know what access is required for those doors specificly but if it's for medical only seems more of a hinderence. The Maintenance rooms in Medical also are not shielded from radiation which seems to be a oversight.

I really like that Toxins Storage now has a window because it brings back the threat of a plasma fire ever since they got nerfed to not burn down walls. Science now has a proper storage room. I don't have any complaints about science I think it's pretty well done.

This might be out of the scope of this PR but is it possible to craft or make electrochromic windows craftable as they become lost forever once someone destroys one?

With that I can't wait for the remap to go through as I enjoyed my time during the testmerge.

**Comment by AffectedArc07** [comment]
A lot of people were reporting pings higher than normal with the testmerge active, which I think may be maptick related, as a lot of these areas are a lot more atom-dense than previous ones, not to mention rampant use of floor decals.

More testing will be done but I have a few concerns.

**Comment by ManiacalSymphony** [comment]
Now that I've played it, the medchem lab could use another window, you can't see who's at the desk while sitting at the lower station

**Comment by AffectedArc07** [comment]
Timeline for what I plan to do with this PR:
1. Move the server to 514 when its stable
2. Add SSperfmon to log maptick, time dilation and other variables over time
3. Gather sample data from an entire day of rounds
4. Testmerge this PR
5. Gather more data from an entire day of rounds
6. Determine whether things are fine

If this PR has major increases to maptick, things will have to change. A **lot** of people are reporting higher pings when running this testmerge, which ties into maptick having to send out more and more data since its a more dense map with 400 decals per tile.

**Comment by ppi13** [comment]
These two rooms can honestly be merged.
And having the APC's in a seperate room from maint it kinda heretical mapping.
image
This room could honestly have that window and door removed
image
28 objects for basically a joke room, is kinda too much, might wanna clean up some of the bits and bobs there.
image
This here room kinda suffers from the same thing, it's overloaded with stuff with 57 objects, giving us a 2.85 items per tiles, so might wanna clean that up
image
So many spawners
image
Experimentor chamber is kinda small, and there is no reason for there to be space near it.
image
Ya want to avoid devices ontop of pipes
image
Same here with devices ontop of pipes
image
And here as well
image
This pipe should be Red
image
Might wanna clean up the catwalks and lattices here, kinda clogged and uneeded.
image
Use fulltile windows here
image
Having these circuits just laying about in maints is not really a great idea, there used to be a chem dispenser there, but it was turned into just a machine frame, so might wanna clear out those, don't want easy access to chemistry for all.
image
Why not have the wheelchair in medical storage instead of in virology
image
Devices ontop of pipes is bad
image
Another room with maybe too many spawners and decals
image
This here is honestly a huge waste of space, that could have gone into making a more spacious exam room, or cryo
image
Where are windows?
image
Four apc in one room? Might wanna put apc into the room where they belong, rather then dedicate a room for them.
image
Also, you got all this big space, and yet this here feels so closed off without any windows
image

**Comment by PacifistDalek** [comment]
Good night, sweet PR prince,
And flights of bridge hobos sing thee to thy rest

**Comment by Spacemanspark** [comment]
Unfortunate.

**Pull request closed without merging**